ballerina/soap_utils.bal

```diff
@@ -1,106 +1,186 @@
+                                       signatureData, serverPublicKey, signatureAlgorithm);
+            }
+        }
+        return soapEnvelope;
+    } on fail var e {
+        return error Error("Outbound security configurations do not match with the SOAP response. ", e.cause());
+    }
+}
+
+string path = "";
+
+public isolated function sendReceive(xml|mime:Entity[] body, http:Client httpClient, string? soapAction = (),
+                                     map<string|string[]> headers = {}, string path = "", boolean soap12 = true) returns xml|Error {
+    http:Request req;
+    if soap12 {
+        req = createSoap12HttpRequest(body, soapAction, headers);
+    } else {
+        req = createSoap11HttpRequest(body, <string>soapAction, headers);
+    }
+    do {
+        http:Response response = check httpClient->post(path, req);
+        if soap12 {
+            return check createSoap12Response(response);
+        }
+        return check createSoap11Response(response);
+    } on fail var err {
+        return error Error(SOAP_RESPONSE_ERROR, err);
+    }
+}
+
+public isolated function sendOnly(xml|mime:Entity[] body, http:Client httpClient, string? soapAction = (),
+                         map<string|string[]> headers = {}, string path = "", boolean soap12 = true) returns Error? {
+    http:Request req;
+    if soap12 {
+        req = createSoap12HttpRequest(body, soapAction, headers);
+    } else {
+        req = createSoap11HttpRequest(body, <string>soapAction, headers);
+    }
+    http:Response|http:ClientError response = httpClient->post(path, req);
+    if response is http:ClientError {
+        return error Error(response.message());
+    }
+}
+
+isolated function createSoap11HttpRequest(xml|mime:Entity[] body, string soapAction,
+                                          map<string|string[]> headers = {}) returns http:Request {
+    http:Request req = new;
+    if body is xml {
+        req.setXmlPayload(body);
+        req.setHeader(mime:CONTENT_TYPE, mime:TEXT_XML);
+    } else {
+        req.setBodyParts(body);
+        req.setHeader(mime:CONTENT_TYPE, mime:MULTIPART_MIXED);
+    }
+    req.addHeader(SOAP_ACTION, soapAction);
+    foreach string key in headers.keys() {
+        req.addHeader(key, headers[key].toBalString());
+    }
+    return req;
+}
+
+isolated function createSoap12HttpRequest(xml|mime:Entity[] body, string? soapAction,
+                                 map<string|string[]> headers = {}) returns http:Request {
+    http:Request req = new;
+    if body is xml {
+        req.setXmlPayload(body);
+    } else {
+        req.setBodyParts(body);
+    }
+    if soapAction is string {
+        map<string> stringMap = {};
+        stringMap[ACTION] = string `${soapAction}`;
+        var mediaType = mime:getMediaType(mime:APPLICATION_SOAP_XML);
+        if mediaType is mime:MediaType {
+            mediaType.parameters = stringMap;
+            req.setHeader(mime:CONTENT_TYPE, mediaType.toString());
+        }

```

Comment:
```suggestion
        var mediaType = mime:getMediaType(mime:APPLICATION_SOAP_XML);
        if mediaType is mime:MediaType {
            mediaType.parameters = {[ACTION]: soapAction};
            req.setHeader(mime:CONTENT_TYPE, mediaType.toString());
        }

```

Reply:
Fixed in [e309edb](https://github.com/ballerina-platform/module-ballerina-soap/pull/103/commits/e309edbbc4d7673f4c2ece8673433cf028487275)